### DISCUSSION TITLE
TOC with chapters and sections

### DISCUSSION THREAD

**Discussion by gdietl** [quarto-dev/quarto-cli#3342] [Q&A] [answered]
I'm currently working on a book using Quarto and struggling with the fact, that the TOC including section, subsections etc. is only visible on wide screens. This makes it very hard to navigate through a document on small screens like, e.g., on mobile phones, where one only sees the overview of the chapters (without sections, subsections etc.). Is there a way to integrate the complete TOC in this overview, i.e., including chapters, sections etc. (similar to the TOC in bookdown)?

**Comment by mcanouil** [comment]
Do you have an example?
Are you talking about the left sidebar (1) or the right sidebar (2)?
<img width="1511" alt="Screenshot 2023-04-06 at 21 55 55" src="https://user-images.githubusercontent.com/8896044/230481641-c03e375f-f1f7-4282-8cb9-f4ac7cbb062a.png">

TOC (2) is in sidebar, on small width screen, you can't have the two columns (Boostrap CSS by the way) and keep the content readable.

It seems you want to include (2) inside (1), but (1) is created from entry in `_quarto.yml` which is parts/chapters defined by "independent" files.
<img width="809" alt="image" src="https://user-images.githubusercontent.com/8896044/230487420-d7a22753-ede6-4fa2-a87a-06ea1c68de67.png">

Note that I don't see what exactly is hard in navigating a book through chapters without sections/subsections, at least for the books in Quarto's website gallery.

**Comment by gdietl** [comment]
Thanks for your comment. Integrating (2) in (1) on small screens is exactly what I would like to have.

If you have chapters with a lot of sections, it is quite cumbersome to swipe through the chapters in order to find a specific section or subsection. One could use the search functionality but then you need to know the names of the sections beforehand. Without the info of (2), you are loosing the information about the structure of the chapter. In fact, this is the only feature from bookdown which I am really missing after switching to quarto. Otherwise, there are so many new cool features why I also do not want to switch back.

**Reply by mcanouil** [reply, marked as answer]
Unfortunately with the current implementation I do not believe there is a way, or at least an easy one, to put (2) in (1).
One workaround would be to tweak css and place the TOC in the body (I did not try).

**Reply by gdietl** [reply]
Thanks for your answer!
